packages/wonder-blocks-form/components/choice-internal.js

```diff
@@ -57,75 +57,82 @@ type Props = {|
  * and RadioGroup. This design allows for more explicit prop typing. For
  * example, we can make onChange a required prop on Checkbox but not on Choice
  * (because for Choice, that prop would be auto-populated by CheckboxGroup).
- */ export default class ChoiceInternal extends React.Component<Props> {
+ */
+export default class ChoiceInternal extends React.Component<Props> {
     static defaultProps = {
         checked: false,
         disabled: false,
         error: false,
     };
 
+    handleLabelClick = (event: SyntheticEvent<>) => {
+        // Browsers automatically use the for attribute to select the input,
+        // but we use ClickableBehavior to handle this.
+        event.preventDefault();
+    };
+
+    handleClick = () => {
+        const {checked, onChange, variant} = this.props;
+        // Radio buttons cannot be unchecked
+        if (variant === "radio" && checked) {
+            return;
+        }
+        onChange(!checked);
+    };
+
     getChoiceCoreComponent() {
         if (this.props.variant === "radio") {
             return RadioCore;
         } else {
             return CheckboxCore;
         }
     }
+
     getLabel() {
         const {disabled, id, label} = this.props;
         return (
             <LabelMedium
                 style={[styles.label, disabled && styles.disabledLabel]}
             >
-                <label
-                    htmlFor={id}
-                    // Browsers automatically use the for attribute to select
-                    // the input, but we use ClickableBehavior to handle this.
-                    onClick={(e) => e.preventDefault()}
-                >
+                <label htmlFor={id} onClick={this.handleLabelClick}>
                     {label}
                 </label>
             </LabelMedium>
         );
     }
+
     getDescription() {
         const {description} = this.props;
         return (
             <LabelSmall style={styles.description}>{description}</LabelSmall>
         );
     }
+
     render() {
         const {
             label,
             description,
+            // eslint-disable-next-line no-unused-vars
             onChange,
             style,
+            // eslint-disable-next-line no-unused-vars
             variant,
```

Comment:
I see, they're in the pre-bound event handler now.  I wish there was a better way to exclude vars from a spread without having unused vars.